thriftbp/server_transport.go

```diff
@@ -0,0 +1,54 @@
+package thriftbp
+
+import (
+	"sync"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/go-kit/kit/metrics"
+
+	"github.com/reddit/baseplate.go/metricsbp"
+)
+
+const meterNameTransportConnCounter = "thrift.connections"
+
+type CountedTServerTransport struct {
+	thrift.TServerTransport
+}
+
+func (m *CountedTServerTransport) Accept() (thrift.TTransport, error) {
+	transport, err := m.TServerTransport.Accept()
+	if err != nil {
+		return nil, err
+	}
+
+	return newCountedTTransport(transport), nil
+}
+
+type countedTTransport struct {
+	thrift.TTransport
+
+	counter   metrics.Counter
+	closeOnce sync.Once
+}
+
+func newCountedTTransport(transport thrift.TTransport) thrift.TTransport {
+	return &countedTTransport{
+		TTransport: transport,
+		counter:    metricsbp.M.Counter(meterNameTransportConnCounter),
+	}
+}
+
+func (m *countedTTransport) Close() error {
+	m.closeOnce.Do(func() {
+		m.counter.Add(-1)
```

Comment:
Sorry I missed this before. for statsd metrics, counter with negative diff is in general a bad idea.

we should use runtime gauge for this instead. gauge also has `Add` function in addition to `Set`: https://pkg.go.dev/github.com/go-kit/kit/metrics#Gauge.Add